src/Agent/NewRelic/Agent/Core/Utilities/HexStringConverter.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+
+namespace NewRelic.Agent.Core.Utilities
+{
+    public static class HexStringConverter
+    {
+        public static byte[] FromHexString(this ReadOnlySpan<char> chars)
+        {
+            if (chars.Length % 2 != 0)
```

Comment:
Is this a general requirement, or is this just applicable to the trace flags scenarios?

Reply:
If it's a general requirement we may want to change the name of this class, because someone might want to use it convert `A` to an array of bytes, because we're naming this class as a "generic" HexStringConverter.

Reply:
It takes two hex characters to (fully) represent one byte. I thought about left-padding with `0` if the length was odd, but decided against it.

Reply:
Should we at least add some `///` comments to call out that limitation? I couldn't come up with a better name.

Reply:
Sure.